tests/framework/extensions/charts/vspherecpi.go

```diff
@@ -0,0 +1,207 @@
+package charts
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rancher/rancher/pkg/api/steve/catalog/types"
+	catalogv1 "github.com/rancher/rancher/pkg/apis/catalog.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/defaults"
+	kubenamespaces "github.com/rancher/rancher/tests/framework/extensions/kubeapi/namespaces"
+	"github.com/rancher/rancher/tests/framework/extensions/namespaces"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+const (
+	// Namespace that rancher cpi chart is installed in
+	VsphereCpiNamespace = "kube-system"
+	// Name of the rancher vpshere cpi chart
+	VsphereCpiName = "rancher-vsphere-cpi"
+)
+
+type CpiConfig struct {
```

Comment:
Nit; can we add godoc comments for these public structs and functions? 

Also for consistency, I think we have these charts opts structures in `charts.go`, can you check `RancherMonitoringOpts` & `RancherIstioOpts` as examples?